docs/docs/examples/bitbucket/BITBUCKET_OPENSEARCH.md

```diff
@@ -0,0 +1,239 @@
+  securityConfig:
+    internalUsersSecret: opensearch-internal-users
+```
+
+If necessary, you can create 2 separate secrets - one for the `internal_users.yml` only, and one with the actual non-hashed OpenSearch credentials that Bitbucket will use.
+
+## Override OpenSearch Helm chart values
+
+You can configure your OpenSearch cluster and the deployment options by overriding any values that the [Helm chart](https://github.com/opensearch-project/helm-charts/blob/main/charts/opensearch/values.yaml){.external} exposes. OpenSearch values must be nested under `opensearch` stanza in your Helm values file, for example:
+
+```yaml
+opensearch:
+  singleNode: false
+  replicas: 5
+  config:
+    opensearch.yml: |
+      cluster.name: custom-cluster
+```
+
+## Enable SSL with Custom Certificates 
+
+By default, OpenSearch starts with the SSL http plugin disabled, meaning the cluster is accessible via HTTP at http://opensearch-cluster-master:9200. The OpenSearch service is not exposed through a LoadBalancer or Ingress unless the default configurations are explicitly overridden. Bitbucket communicates with the OpenSearch cluster using the service name within the internal Kubernetes network. This setup uses the Kubernetes DNS to resolve the service name to the appropriate cluster IP address.
+
+To enable SSL in OpenSearch and start the service on a secure port, you need to:
+* enable ssl http in Helm values
+* create Kubernetes secrets with ca, certificate and key, and pass them to OpenSearch
+* add ca.crt to Java trust store in Bitbucket containers if the custom certificate is not signed by a public authority
+
+Below is an example of how to generate a CA certificate, a server certificate, and a corresponding private key for securing communications with OpenSearch:
+
+```shell
+openssl req -new -newkey rsa:2048 -days 365 -nodes -x509 -subj "/C=US/ST=CA/L=LA/O=MyCompany Name/CN=opensearch-cluster-master" -keyout ca.key -out ca.crt
+openssl req -new -newkey rsa:2048 -nodes -keyout opensearch-cluster-master.key -subj "/C=US/ST=CA/L=LA/O=MyCompany Name/CN=opensearch-cluster-master" -out opensearch-cluster-master.csr
+openssl x509 -req -days 365 -in opensearch-cluster-master.csr -CA ca.crt -CAkey ca.key -CAcreateserial -out opensearch-cluster-master.crt
+```
+
+It is important to generate the certificate for `opensearch-cluster-master` CN because this is the actual OpenSearch service/host name in the target namespace.
+
+Once done, create 3 Kubernetes secrets:
+
+```shell
+kubectl create secret generic opensearch-ssl-ca -n atlassian --from-file=root-ca.pem=ca.crt
+kubectl create secret generic opensearch-ssl-key -n atlassian --from-file=esnode-key.pem=opensearch-cluster-master.key
+kubectl create secret generic opensearch-ssl-cert -n atlassian --from-file=esnode.pem=opensearch-cluster-master.crt
+```
+
+Enable ssl http plugin, override the default ca, certificate and key, as well as provide additional volume mounts:
+
+```yaml
+opensearch:
```

Comment:
I think it is good to clarify that these changes need to be made to the values of the bitbucket helm chart.